plugins/ua_pubsub_ethernet.c

```diff
@@ -1247,7 +1247,7 @@ UA_PubSubChannelEthernet_receive(UA_PubSubChannel *channel,
         buffer.length = messageLength;
 
         retval = receiveCallback(channel, receiveCallbackContext, &buffer);
-        if (retval != UA_STATUSCODE_GOOD) {
+        if (UA_StatusCode_isBad(retval)) {
```

Comment:
Sorry my mistake
This change has nothing to do with the publisherId implementation.
It should have been a separate commit.

If we have multiple connections within the same multicast group (as unit-test `Test_multiple_connections` checks) then every connection calls receive() and gets the messages from all publishers.
So 1 subscriber receives for example 6 messages, but only 1 of them is the interesting one and all the others are for another Subscriber.

If the message is not for the Subscriber then we get following messages if implementation is still retval != UA_STATUSCODE_GOOD

```

[32minfo/server[0m	PubSub receive. Unknown message received. Will not be processed.
[33mwarn/network[0m	PubSub Connection decode and process failed.
```

And somehow I didn't like the warning: PubSub Connection decode and process failed.
Because it looked like something is wrong but in reality the received message was just for another subscriber.
I wanted to only print the message if there was a "real" error, e.g. decoding failed or any other internal programming error.

If I change the implementation to (UA_StatusCode_isBad(retval))
then I only get 

`[32minfo/server[0m	PubSub receive. Unknown message received. Will not be processed.`

```

receiveCallback return values can be
   GOOD, BAD, UNCERTAIN
   
   decodeAndProcessFunRT
      decodeAndProcessNetworkMessageRT
         UA_NetworkMessage_decodeHeaders
         verifyAndDecryptNetworkMessage

         UA_NetworkMessage_updateBufferedNwMessage
            UA_STATUSCODE_UNCERTAIN
            
         checkReaderIdentifier() - would return UA_STATUSCODE_BADNOTFOUND
               but return value is set to uncertain
               UA_STATUSCODE_UNCERTAIN
   
   decodeAndProcessFun
      decodeAndProcessNetworkMessage
         decodeNetworkMessage
         UA_Server_processNetworkMessage -> returns always GOOD
            checkReaderIdentifier

```

Reply:
But maybe warning is not that bad, because the subscriber actually couldn't process message, and it's not an error, so maybe I should not have changed it

I have removed it for now